### PR TITLE
[JSON language] add diagnostic codes to ignore

### PR DESCRIPTION
**Pull request by afvieira**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
 	"name": "@outsystems/monaco-editor",
-	"version": "0.31.1-os",
+	"version": "0.31.4-os",
```
Comment (os-lmo) [inline package.json]: Why 4? I would suggest `0.31.1-os-2`

File: src/json/monaco.contribution.ts [hunk 2]
```diff
@@ -109,6 +109,11 @@ export interface ModeConfiguration {
 	 */
 	readonly diagnostics?: boolean;
 
+	/**
+	 * Defines diagnostic codes to be ignored.
+	 */
+	readonly diagnosticCodesToIgnore?: number[];
```
Comment (os-lmo) [inline src/json/monaco.contribution.ts]: Why don't this diagnosticCodesToIgnore live under a `diagnosticsoptions` object such as the Typescript/Javascript implementation?
https://blutorange.github.io/primefaces-monaco/typedoc/interfaces/monaco.languages.typescript.diagnosticsoptions.html

File: src/common/lspLanguageFeatures.ts [hunk 3]
```diff
@@ -105,7 +106,10 @@ export class DiagnosticsAdapter<T extends ILanguageWorkerWithDiagnostics> {
 			.then((worker) => {
 				return worker.doValidation(resource.toString());
 			})
-			.then((diagnostics) => {
+			.then((allDiagnostics) => {
+				const diagnostics = allDiagnostics
+						.filter(diagnostic => (this._diagnosticCodesToIgnore || []).indexOf(diagnostic.code) === -1);
```
Comment (os-lmo) [inline src/common/lspLanguageFeatures.ts]: I would suggest moving this check `this._diagnosticCodesToIgnore || []` to the constructor (since it's the unique place where it can be assigned).
Hence, this line can be:
```const diagnostics = allDiagnositcs.filter(diagnostic => diagnosticCodesToIgnore.includes(diagnostic.code));```

File: src/common/lspLanguageFeatures.ts [hunk 4]
```diff
@@ -35,7 +35,8 @@ export class DiagnosticsAdapter<T extends ILanguageWorkerWithDiagnostics> {
 	constructor(
 		private readonly _languageId: string,
 		protected readonly _worker: WorkerAccessor<T>,
-		configChangeEvent: IEvent<any>
+		configChangeEvent: IEvent<any>,
+		protected readonly _diagnosticCodesToIgnore: number[]
```
Comment (os-lmo) [inline src/common/lspLanguageFeatures.ts]: Can you add a default value here? e.g. Empty array

File: src/css/cssMode.ts [hunk 5]
```diff
@@ -95,7 +95,7 @@ export function setupMode(defaults: LanguageServiceDefaults): IDisposable {
 		}
 		if (modeConfiguration.diagnostics) {
 			providers.push(
-				new languageFeatures.DiagnosticsAdapter(languageId, worker, defaults.onDidChange)
+				new languageFeatures.DiagnosticsAdapter(languageId, worker, defaults.onDidChange, defaults.modeConfiguration?.diagnosticCodesToIgnore)
```
Comment (os-lmo) [inline src/css/cssMode.ts]: You don't need this feature in CSS, right? If true, consider to undo these changes and open a PR to the official repo with both changes.

**Review by os-lmo — COMMENTED** [review]
See my comments please.
Besides, please rename the PR to mention explicitly the JSON language

File: src/common/lspLanguageFeatures.ts (hunk 3 above)
Comment (afvieira) [reply src/common/lspLanguageFeatures.ts]: in this case we want to exclude all the ones that I want to ignore. `includes` does something different, right? :p

Comment (afvieira) [reply src/common/lspLanguageFeatures.ts]: example: 
image

File: src/common/lspLanguageFeatures.ts [hunk 6]
```diff
@@ -105,7 +106,10 @@ export class DiagnosticsAdapter<T extends ILanguageWorkerWithDiagnostics> {
 			.then((worker) => {
 				return worker.doValidation(resource.toString());
 			})
-			.then((diagnostics) => {
+			.then((allDiagnostics) => {
+				const diagnostics = allDiagnostics
+						.filter(diagnostic => this._diagnosticCodesToIgnore.indexOf(diagnostic.code) === -1);
```
Comment (tbragaf) [inline src/common/lspLanguageFeatures.ts:111]: can you use !includes?

**Review by tbragaf — APPROVED** [review]

File: src/common/lspLanguageFeatures.ts (hunk 6 above)
Comment (JMano) [reply src/common/lspLanguageFeatures.ts:111]: I'd say we can. I'm having some difficulties building this but I'll commit that change and trigger a test run in the pipeline

Comment (JMano) [reply src/common/lspLanguageFeatures.ts:111]: Ended up not using because it was meant to use as it is due to being consistent with the code from the Typescript language, for example.
